app/components/Texting/Texting.jsx

```diff
@@ -0,0 +1,223 @@
+          <div className="dataRates">*Standard text and data rates may apply.</div>
+          <input
+            type="text"
+            name="phoneNumber"
+            className="input"
+            value={phoneNumber}
+            data-field="phoneNumber"
+            onChange={onChange}
+          />
+        </div>
+
+        <label className="checkBox dataRates">
+          I agree to receive text messages from SF Service Guide.
+          <input
+            type="checkbox"
+            name="agreed"
+            checked={agreed}
+            data-field="agree"
+            onChange={onChange}
+          />
+        </label>
+      </div>
+
+      <div className="buttons">
+        <button
+          type="button"
+          className="cancel button"
+          onClick={toggle}
+        >
+          cancel
+        </button>
+        <button
+          type="submit"
+          className={`button${agreed && phoneNumberValidator(phoneNumber) ? '' : ' disabled'}`}
+          disabled={!(agreed && phoneNumberValidator(phoneNumber))}
+          data-field="submit"
+          onClick={onSubmit}
+        >
+          Text me
+        </button>
+      </div>
+
+      <div className="privacy">
+        We will not share or sell your information with third party sites.
+        For more information on privacy details, please go
+        <a target="_blank" rel="noopener noreferrer" href="https://sfserviceguide.org/"> here </a>
+        to view.
+      </div>
+    </>
+  );
+
+  // Loading spinner: Show a loading spinner when the request being processed in the Backend
+  const loadingView = () => (<Loader />);
+
+  // Success sending view
+  const sentView = () => (
+    <div className="responseIcons">
+      <div className="circle">
+        <img src={images.icon('check-mark')} alt="check-mark" className="responseIcon" />
+        <h1 className="responseText" data-field="sent">Sent!</h1>
+      </div>
+    </div>
+  );
+
+  // Error View: show Alert graphic and a friendly message
+  const errorView = () => (
+    <div className="responseIcons error">
+      <img src={images.icon('emergency')} alt="error" className="responseIcon" />
+      <h1 className="responseText error">Sorry, something went wrong, please try again later.</h1>
+    </div>
+  );
+
+  const view = () => {
+    if (error) {
+      return errorView();
+    }
+    if (isSent) {
+      return sentView();
+    }
+    if (isLoading) {
+      return loadingView();
+    }
+    return formView();
+  };
+  // create a portal to render Texting component on Top of the DOM hierarchy
+  return (
+    ReactDOM.createPortal(
+      <div className="overlay">
+        <div className="popup">
+          <div className="closeIconDiv" role="button" onClick={toggle} tabIndex={0}>
+            <img src={images.icon('close')} alt="close" className="closeIcon" />
+          </div>
+          { view() }
+        </div>
+      </div>, document.body,
+    )
+  );
+};
```

Comment:
This is quite a long function, and it might be a good idea to split it up into different pieces. One possible way of organizing this would be split the "logic" from the "view", where the state and the callbacks can be defined in one component and the actual view can be implemented as a pure function component. Another way of organizing the component would be to split up the views into separate smaller components representing the individual views.